src/realm/object-store/sync/sync_session.cpp

```diff
@@ -885,16 +892,30 @@ void SyncSession::create_sync_session()
     }
 
     {
-        std::string sync_route = m_sync_manager->sync_route();
+        // At this point, the sync_route should be valid, since the location should have been updated by
+        // an AppServices http request or by updating the access token before starting the sync session.
+        auto sync_route = m_sync_manager->sync_route();
+        // If sync_route is not valid at this point, it means the location update failed - pass the error up via
+        // the registered error handler
+        if (!sync_route) {
```

Comment:
If m_session was created above, doesn't this mean the sync_route should always exist (or throw above otherwise)? if anything we should assert that sync_route always exists here.